sdk/tests/client/input_selection/account_outputs.rs

```diff
@@ -1644,3 +1683,191 @@ fn remainder_address_in_state_controller() {
+        ManaAllotment::new(account_id_1, mana_cost).unwrap()
+    );
+    assert_eq!(
+        selected.transaction.outputs()[1].as_account().mana(),
+        mana_input_amount - mana_cost
+    );
+}
+
+#[test]
+fn automatic_allot_account_mana_additional() {
+    let protocol_parameters = protocol_parameters();
+    let account_id_1 = AccountId::from_str(ACCOUNT_ID_1).unwrap();
+
+    let additional_allotment = 1000;
+    let txn_required_mana_allotment = 240_000;
+    // The account does not have enough to cover the requirement
+    let account_mana = txn_required_mana_allotment - 100;
+    // But there is additional available mana elsewhere
+    let additional_available_mana = additional_allotment + 111;
+
+    let inputs = [
+        AccountOutputBuilder::new_with_amount(2_000_000, account_id_1)
+            .add_unlock_condition(AddressUnlockCondition::new(
+                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+            ))
+            .with_mana(account_mana)
+            .finish_output()
+            .unwrap(),
+        BasicOutputBuilder::new_with_minimum_amount(protocol_parameters.storage_score_parameters())
+            .with_mana(additional_available_mana)
+            .add_unlock_condition(AddressUnlockCondition::new(
+                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+            ))
+            .finish_output()
+            .unwrap(),
+    ];
+    let inputs = inputs
+        .into_iter()
+        .map(|input| InputSigningData {
+            output: input,
+            output_metadata: rand_output_metadata_with_id(rand_output_id_with_slot_index(SLOT_INDEX)),
+            chain: None,
+        })
+        .collect::<Vec<_>>();
+
+    let outputs = [BasicOutputBuilder::new_with_amount(1_000_000)
+        .with_mana(11)
+        .add_unlock_condition(AddressUnlockCondition::new(
+            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+        ))
+        .add_feature(SenderFeature::new(
+            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+        ))
+        .finish_output()
+        .unwrap()];
+
+    let selected = InputSelection::new(
+        inputs.clone(),
+        outputs.clone(),
+        [Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap()],
+        SLOT_INDEX,
+        SLOT_COMMITMENT_ID,
+        protocol_parameters,
+    )
+    .with_auto_mana_allotment(account_id_1, 500)
+    .with_mana_allotments(Some((account_id_1, additional_allotment)))
+    .select()
+    .unwrap();
+
+    assert!(unsorted_eq(&selected.inputs_data, &inputs));
+    assert_eq!(selected.transaction.outputs().len(), 2);
+    assert!(selected.transaction.outputs().contains(&outputs[0]));
+
+    assert_eq!(selected.transaction.allotments().len(), 1);
+    assert_eq!(
+        selected.transaction.allotments()[0],
+        ManaAllotment::new(account_id_1, txn_required_mana_allotment + additional_allotment).unwrap()
```

Comment:
if both manual and auto allotment are provided for the same account, should we not cap it to the auto value if it's more than the manual value?

Reply:
That's how it was before, but I think if you provide an amount manually you probably don't want that amount reduced